pkg/helmcli/adder.go

```diff
@@ -0,0 +1,134 @@
+package helmcli
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gojekfarm/albatross/pkg/helmcli/flags"
+	"github.com/gojekfarm/albatross/pkg/logger"
+
+	"github.com/gofrs/flock"
+	"gopkg.in/yaml.v2"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const timeout time.Duration = 30 * time.Second
+
+type adder struct {
+	flags.AddFlags
+	settings *cli.EnvSettings
+}
+
+func (o *adder) Add(ctx context.Context) error {
+	err := o.checkPrerequisite()
+	if err != nil {
+		return err
+	}
+
+	// Acquire a file lock for process synchronization
+	fileLock := flock.New(strings.Replace(o.RepoFile, filepath.Ext(o.RepoFile), ".lock", 1))
+	lockCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	locked, err := fileLock.TryLockContext(lockCtx, time.Second)
+	if err == nil && locked {
+		defer check(fileLock.Unlock)
```

Comment:
Why is this abstracted out instead of directly calling Unlock() here in defer

Reply:
For logging purposes, if the unlock returns an error it should be logged.